.github/workflows/playground_deploy_examples.yml

```diff
@@ -26,23 +25,33 @@ env:
   BEAM_VERSION: 2.40.0
   K8S_NAMESPACE: playground-backend
   HELM_APP_NAME: playground-backend
+permissions: read-all
 jobs:
   check_examples:
     name: Check examples
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, ubuntu-20.04]
     outputs:
       example_has_changed: ${{ steps.check_has_example.outputs.example_has_changed }}
     steps:
       - name: Check out the repo
         uses: actions/checkout@v3
         with:
           fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
       - uses: actions/setup-python@v4
         with:
           python-version: '3.8'
       - name: install deps
         run: pip install -r requirements.txt
         working-directory: playground/infrastructure
+      - name: Install PowerShell
```

Comment:
Rather than installing powershell, can we just convert the `pwsh` block below to a more generic script block? It isn't doing anything that a bash shell shouldn't be able to do.

Reply:
In general, we are adding the required installations to make the workflows work. In this case, we are afraid that by changing this block of code into bash shell the job may get broken. 

As we don't really know the reason behind this PowerShell decision, we suggest to create a new task/issue for this change. 

Please let us know your comments. Thanks!

Reply:
> we suggest to create a new task/issue for this change.

That's fine by me, this doesn't need to be blocking